src/bun.js/net.exports.js

```diff
@@ -368,7 +380,206 @@ export function createConnection(port, host, connectListener) {
 
 export const connect = createConnection;
 
+class Server extends EventEmitter {
+  #connectionListener;
+  #options;
+  #server;
+  #listening;
+  #connections;
+
+  constructor(options, connectionListener) {
+    super();
+    this.maxConnections = 0;
+
+    if (typeof options === "function") {
+      connectionListener = options;
+      options = {};
+    } else if (options == null || typeof options === "object") {
+      options = { ...options };
+    } else {
+      throw new Error("bun-net-polyfill: invalid arguments");
+    }
+
+    if (typeof options.maxConnections === "number" && options.maxConnections > 0) {
```

Comment:
This calls the getter three times and doesn't handle

Instead, something like this would be better
```js
const {maxConnections} = options;
this.maxConnections = Number.isSafeInteger(maxConnections) && maxConnections > 0 ? maxConnections : 0;
```

Reply:
done on https://github.com/oven-sh/bun/pull/2337/commits/dc6bc3c6d8d73d3f5588116eeeaab409d01c765f